### PR TITLE
makes syscall openat behave like syscall open

### LINKED ISSUE #72: syscall openat didnt create file when O_CREAT is passed
syscall openat returns "File Not Found" but not create it even O_CREAT is passed

### PR DESCRIPTION
**Pull request by ucgJhe**
fix #72 and add test for x86_linux

### CODE REVIEW

File: qiling/os/posix/syscall.py [hunk 1]
```diff
@@ -252,24 +251,29 @@ def ql_syscall_openat(ql, openat_fd, openat_path, openat_flags, openat_mode, nul
     openat_flags = openat_flags & 0xffffffff
     openat_mode = openat_mode & 0xffffffff
 
-    if os.path.exists(real_path) == False:
+    for i in range(256):
+        if ql.file_des[i] == 0:
+            idx = i
+            break
+
+    if idx == -1:
```
Comment (cq674350529) [inline qiling/os/posix/syscall.py]: Does it need to initialize `idx` with `-1` first?

Comment (ucgJhe) [reply qiling/os/posix/syscall.py]: u r right, fixed it with an extra `else` to do that

Comment (cq674350529) [reply qiling/os/posix/syscall.py]: The `else` statement added is odd: missing indentation.
By the way, maybe you can refer to this:
https://github.com/qilingframework/qiling/blob/6596ceb1997adbff04f7add1f7012ba7d6adbb65/qiling/os/posix/syscall.py#L1242-L1254

Comment (ucgJhe) [reply qiling/os/posix/syscall.py]: the `else` here is paired with `for` statement so the indentation is fine

Comment (cq674350529) [reply qiling/os/posix/syscall.py]: Sorry for that. Cool! Did't notice this before ...

Comment (ucgJhe) [reply qiling/os/posix/syscall.py]: its okay, im glad you spot the oddity of idx value thing. thx

**Comment by xwings** [comment]
Hi,

Thanks !

**Pull request merged**